point_pattern.py

```diff
@@ -120,7 +144,16 @@ def average_nearest_neighbor_distance(points):
      p. 445-453.
     """
     mean_d = 0
-
+    for i in points:
+        dist_nearest=1e9
+        for j in points:
+            dist = euclidean_distance(i, j)
+            if i==j:
```

Comment:
Check out `enumerate`.  It will let you know what positional index the iterator is at in the list.  You can use this to check if both iterators are in the same place - thereby not skipping coincident points.